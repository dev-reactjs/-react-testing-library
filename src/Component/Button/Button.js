import React from "react"
function Button({ label }) {
    return (
        <div data-testid="button" className="button-style">
           Hii {label}
        </div>
    );
}

export default Button;
