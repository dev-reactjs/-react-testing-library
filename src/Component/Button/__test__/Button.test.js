import React from "react"
import Button from './../Button';
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"
afterEach(cleanup)
it('renders without crashing', () => {
    const div = document.createElement("div")
    ReactDom.render(<Button></Button>, div)
});

it('renders button correctly', () => {
    const { getByTestId } = render(<Button label="Click me please"></Button>)
    expect(getByTestId("button")).toHaveTextContent("Click me please");
});
it('renders button correctly', () => {
    const { getByTestId } = render(<Button label="save"></Button>)
    expect(getByTestId("button")).toHaveTextContent("save");
});

it("matches snapshot 1",()=>{
    const tree =renderer.create(<Button label="save"></Button>).toJSON()
    expect(tree).toMatchSnapshot();
})

it("matches snapshot 2",()=>{
    const tree =renderer.create(<Button label="Click me please"></Button>).toJSON()
    expect(tree).toMatchSnapshot();
})