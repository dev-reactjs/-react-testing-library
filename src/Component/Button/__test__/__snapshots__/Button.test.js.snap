// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hii 
  save
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hii 
  Click me please
</div>
`;
